main/src/cgeo/geocaching/filter/CountryFilterFactory.java

```diff
@@ -0,0 +1,84 @@
+package cgeo.geocaching.filter;
+
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.storage.DataStore;
+
+import android.os.Parcel;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CountryFilterFactory implements IFilterFactory {
+
+
+    static class CountryFilter extends AbstractFilter {
+        private String location = null;
+
+        public static final Creator<CountryFilter> CREATOR
+                = new Creator<CountryFilter>() {
+
+            @Override
+            public CountryFilter createFromParcel(final Parcel in) {
+                return new CountryFilter(in);
+            }
+
+            @Override
+            public CountryFilter[] newArray(final int size) {
+                return new CountryFilter[size];
+            }
+        };
+
+        CountryFilter(final String location, final String title) {
+            super(title);
+            this.location = location;
+        }
+
+        protected CountryFilter(final Parcel in) {
+            super(in);
+            location = in.readString();
+        }
+
+        @Override
+        public void writeToParcel(final Parcel dest, final int flags) {
+            super.writeToParcel(dest, flags);
+            dest.writeString(location);
+        }
+
+        @Override
+        public boolean accepts(@NonNull final Geocache cache) {
+            final @NonNull String location = cache.getLocation();
+
+            return location != null && location.endsWith(this.location);
+        }
+
+    }
+
+    @Override
+    @NonNull
+    public List<IFilter> getFilters() {
+
+        final Map<String, IFilter> filters = new HashMap<>();
+
+        final String separator = ", ";
+
+        // TODO: use DataStore.getStoredLocationsForSet(Set<String>)
```

Comment:
Here, I need to somehow get the list of current geocodes.

Reply:
@moving-bits any idea?

Reply:
@jakedot  Nothings springs to my mind immediately, but I haven't looked into the source code yet. Maybe have a look at the position where the filter gets instantiated - can you add the set as a new parameter there?